src/main/scala/ReproducibleBuildsPlugin.scala

```diff
@@ -101,12 +95,62 @@ object ReproducibleBuildsPlugin extends AutoPlugin {
       reproducibleBuildsPackageName.value,
       ext,
       ext,
-      "compile",
+      Compile.name,
       extraModuleAttributes,
       null
     )
   }
 
+  def locationFromResolver(r: DependencyResolver, ext:String) = Def.task {
+    import SbtLibraryManagementFunctions._
+    val moduleInfo = (moduleSettings.value match {
+      case ic: InlineConfiguration =>
+        val icWithCross = substituteCross(ic)
+        if (sbtPlugin.value) appendSbtCrossVersion(icWithCross)
+        else icWithCross
+    }).moduleInfo
+
+    val artifactOrigin = r.locate(
+      toIvyArtifact(
+        newConfiguredModuleID(moduleID.value, moduleInfo, Vector(Compile)),
+        artifact.value.withExtension(ext),
+        Vector(Compile)
+      )
+    )
+    artifactOrigin.getLocation
+  }
+
+  def artifactUrl(resolver: Resolver, ext: String) = Def.taskDyn {
+    import SbtLibraryManagementFunctions._
+
+    resolver match {
+      case repository: MavenRepository =>
+        val pattern = resolvePattern(
```

Comment:
This is copied from https://github.com/sbt/librarymanagement/blob/10a79d6001a6c43a9cdf66a7c5c96748222c8f61/core/src/main/scala/sbt/librarymanagement/ResolverExtra.scala#L386-L390 , it properly handles combining a base pattern along with a root repository URIL (as an example it handles the trailing `/`).

This is what allows sbt-reproducible-builds to handle generic `Resolver`'s